externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/PlanOnlyStatement.scala

```diff
@@ -59,7 +59,7 @@ class PlanOnlyStatement(
   override protected def resultSchema: StructType = {
     if (result == null) {
       new StructType().add("plan", "string")
-    } else if (result.isEmpty) {
+    } else if (result.schema.isEmpty) {
```

Comment:
If I ger it correctly, when we go into thie else if branch, that means the query is in `planExcludes` ? We return the result rather than the plan. Can we add a new flag to distinguish them ? I'm not sure if the result schema of `planExcludes` is always empty.

Reply:
I think we don't need to add a flag to distinguish them, because we handle all cases：

```
if (result == null) {
    ...
else if (result.schema.isEmpty) {
    ...
} else {
    result.schema
}
```

Reply:
And it is consistent with `ExecuteStatement`

https://github.com/apache/kyuubi/blob/635c79373df921018a48cb7c354719e5a9254641/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala#L53-L58

Reply:
ah, I see. How about reduce the else if and else by using `super.resultSchema()` ?

Reply:
Sounds good, thanks, I will change that